elastic/kibana
# [ResponseOps][Alerting] ESQL alerting rule type

Resolves https://github.com/elastic/kibana/issues/153448

## Summary

This PR adds ESQL query type to the ES query rule. The feature is hidden by the `discover:enableTextBased` and can be enabled in the advanced settings. The ESQL query type will alert on 'query matched' if any documents are returned from the query and recover when no documents are returned.

Unlike the POC, ESQL now queries on indices instead of data views, so users now need to choose a time field. It will default to @ timestamp is available, otherwise, the user will need to select a field.


### Checklist



### To verify

- Create a new ES Query rule and verify that ESQL is listed as an option.
- Create a new rule and verify that it works as expected.
- Go to advanced settings and disable `discover: enableESQL `
- Go back to create an ES Query rule and verify that ESQL is not listed
